crates/guest/assessor/assessor-guest/src/main.rs

```diff
@@ -53,8 +54,17 @@ fn main() {
     // - record the selector if it is present
     // We additionally collect the request and claim digests.
     for (index, fill) in input.fills.iter().enumerate() {
-        let request_digest =
-            fill.verify_signature(&eip_domain_separator).expect("signature does not verify");
+        // ECDSA signatures are always checked here.
+        // Smart contract signatures (via EIP-1271) are checked on-chain either when a request is locked,
+        // or when an unlocked request is priced and fulfilled.
+        let smart_contract_signed = RequestId::from_lossy(fill.request.id).smart_contract_signed;
```

Comment:
In this case, I think we should panic on unknown flags, rather than ignoring them. It's a balance between forwards-compatibility and preventing accidental issues. In the guess, I'd favor the stricter approach.
```suggestion
        let smart_contract_signed = RequestId::try_from(fill.request.id).unwrap().smart_contract_signed;
```

Reply:
212b74d11184a83b76211c20aa1a06616c2f7d27